ui/pages/confirmations/components/simulation-details/types.ts

```diff
@@ -36,57 +36,28 @@ export type AssetIdentifier = Readonly<
 >;
 
 /**
- * Represents an amount of an asset, including its magnitude and sign.
+ * Describes a change in an asset's balance to a user's wallet.
  */
-export type Amount = Readonly<{
-  /**
-   * Indicates whether the amount is negative (e.g., a decrease in balance).
-   */
-  isNegative: boolean;
-
+export type BalanceChange = Readonly<{
   /**
-   * The quantity of the smallest denomination of the asset (base units),
-   * represented as a hexadecimal string.
-   * For example: In the case of ETH, this would be the number of wei.
+   * The asset identifier for the balance change.
    */
-  quantity: Hex;
+  asset: AssetIdentifier;
 
   /**
-   * The number of decimal places the associated asset supports.
+   * The quantity of asset tokens, expressed as a decimal value.
    *
-   * This value is the negation of the exponent used when converting
-   * the quantity to the decimal amount of a token.
+   * This property represents the amount of tokens, taking into account the
+   * number of decimals supported by the asset. The value can be positive
+   * (increase) or negative (decrease).
    *
-   * To calculate the token amount in decimal form, use the formula:
-   * `tokenAmount = hexToDecimal(quantity) / (10 ^ decimals)`
-   *
-   * Example: If the asset is ETH, the quantity is expressed in wei
-   * (the smallest unit of ETH) and decimals would be 18. The amount
-   * of ETH tokens would be: `ethAmount = quantity / (10 ^ 18)`
+   * Example: If an asset supports 18 decimals, an `amount` of 1.5 represents
+   * 1.5 tokens, or more precisely, 1.5 * 10^18 of the smallest divisible unit.
    */
-  decimals: number;
-
-  /**
-   * The numeric representation of the amount, taking into account the
-   * sign, quantity and decimals.
-   */
-  numeric: Numeric;
```

Comment:
Since `Numeric` uses `BigNumber.js` internally, is there an argument to still use `Numeric` since it provides an extension of the `BigNumber.js` functionality for easier conversion between ETH denominations for example?